pkg/server/util/regentryutil/fetch.go

```diff
@@ -25,8 +33,14 @@ func newRegistrationEntryFetcher(dataStore datastore.DataStore) *registrationEnt
 	}
 }
 
-func (f *registrationEntryFetcher) Fetch(ctx context.Context, id string) ([]*common.RegistrationEntry, error) {
-	entries, err := f.fetch(ctx, id, make(map[string]bool))
+func (f *registrationEntryFetcher) Fetch(ctx context.Context, cache *FetchRegistrationEntriesCache, id string) ([]*common.RegistrationEntry, error) {
+	var entries []*common.RegistrationEntry
+	var err error
+	if cache != nil {
```

Comment:
same here regarding switching on existence of the cache

Reply:
There are actually other APIs like `fetchJWTSVID` and `Attest` which use non cache fetch.

Reply:
I think you could get rid of most of the newly-added branching in this package if the cache passed to`registrationEntryFetcher` was an (unexported) interface, and in the case where caching wasn't taking place, it could be passed a `nullCache` that would never return anything from `Get` and ignore calls to `Set`.

Reply:
Thank you, it has become much clean now!